From c877d62a4a87f40f64facaed0c24dd5bda44ea27 Mon Sep 17 00:00:00 2001
From: Yida Zhang <hitzhangyd@gmail.com>
Date: Wed, 14 Nov 2012 11:37:02 -0500
Subject: [PATCH 3/3] add NO_CMAKE_FIND_ROOT_PATH to rospack search

---
 core/rosbuild/rosbuild.cmake |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/rosbuild/rosbuild.cmake b/core/rosbuild/rosbuild.cmake
index 10629a5..899b70e 100644
--- a/core/rosbuild/rosbuild.cmake
+++ b/core/rosbuild/rosbuild.cmake
@@ -28,7 +28,8 @@ set(CMAKE_OSX_ARCHITECTURES "x86_64")
 
 ##############################################################################
 # First things first: we must have rospack.  But only look at PATH, #3831
-find_program(ROSPACK_EXE NAMES rospack DOC "rospack executable" NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH)
+find_program(ROSPACK_EXE NAMES rospack DOC "rospack executable" NO_CMAKE_PATH 
+              NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
 if (NOT ROSPACK_EXE)
   message(FATAL_ERROR "Couldn't find rospack. Please run 'make' in $ROS_ROOT")
 endif(NOT ROSPACK_EXE)
-- 
1.7.9.5

